decidim-core/lib/decidim/core/test/shared_examples/map_examples.rb

```diff
@@ -57,10 +57,17 @@ def snippets
 
 shared_context "with frontend map elements" do
   let(:html_head) { "" }
-  let(:html_document) do
+  let(:builder_snippets) do
     builder = subject
+    <<~HTML.strip
+      #{builder.stylesheet_snippets}
```

Comment:
Actually the problem here is the fact that we call those append_javascript_pack_tag, and yet, we do not call the javascript_pack_tag. 

Also, shakapacker is asking you to call the append_* before calling the actual *_pack_tag. The behaviour is like this, as in Rails, the partials are being compiled before layouts.

Reply:
OK, so then we can remove this with the new approach right?

Because the tags are generated when `html_body` is called and therefore the JS/CSS injections should be already in the "queue" when we are calling the `x_pack_tag` methods in the template, right?

Reply:
I have made an exercise, and i have loaded 2 maps in the Meetings extras like : 

```
<% if current_component.settings.maps_enabled? && search.result.not_online.exists? %>
  <%= cell "decidim/meetings/meetings_map", search.result %>
  <%= cell "decidim/meetings/meetings_map", search.result %> # added for testing 
<% end %>
```

and the resulted JS was : 

```
<script src="/decidim-packs/js/vendors-node_modules_leaflet_markercluster_dist_leaflet_markercluster-src_js-node_modules_lea-91385c.js" defer="defer"></script>
<script src="/decidim-packs/js/decidim_decidim-core_app_packs_src_decidim_map_js.js" defer="defer"></script>
<script src="/decidim-packs/js/decidim_map_provider_here.js" defer="defer"></script>
<script src="/decidim-packs/js/vendors-node_modules_webpack-dev-server_client_index_js_protocol_ws_3A_hostname_localhost_por-cf1095.js"></script>
<script src="/decidim-packs/js/vendors-node_modules_quill_formats_code_js.js"></script>
<script src="/decidim-packs/js/vendors-node_modules_tarekraafat_autocomplete_js_dist_autoComplete_min_js-node_modules_tarekr-f21bdb.js"></script>
<script src="/decidim-packs/js/vendors-node_modules_rails_activestorage_app_assets_javascripts_activestorage_js-node_modules-03f302.js"></script>
<script src="/decidim-packs/js/vendors-node_modules_d3-array_src_index_js-node_modules_d3-collection_src_index_js-node_modul-f618e0.js"></script>
<script src="/decidim-packs/js/vendors-node_modules_d3-axis_src_index_js-node_modules_d3-scale_src_index_js-node_modules_d3--9942d3.js"></script>
<script src="/decidim-packs/js/vendors-node_modules_zeitiger_appendaround_appendAround_js-node_modules_a11y-accordion-compon-1b82f6.js"></script>
<script src="/decidim-packs/js/decidim_decidim-core_app_packs_src_decidim_editor_js.js"></script>
<script src="/decidim-packs/js/decidim_decidim-core_app_packs_src_decidim_autocomplete_js.js"></script>
<script src="/decidim-packs/js/decidim_decidim-core_app_packs_src_decidim_configuration_js-decidim_decidim-core_app_packs_sr-2a2c67.js"></script>
<script src="/decidim-packs/js/decidim_decidim-core_app_packs_src_decidim_form_datepicker_js-decidim_decidim-core_app_packs_-d477e6.js"></script>
<script src="/decidim-packs/js/decidim_decidim-core_app_packs_src_decidim_direct_uploads_upload_field_js.js"></script>
<script src="/decidim-packs/js/decidim_decidim-core_app_packs_images_sync_recursive_-decidim_decidim-core_app_packs_src_deci-80e6ef.js"></script>
<script src="/decidim-packs/js/decidim_decidim-core_app_packs_src_decidim_ajax_modals_js-decidim_decidim-core_app_packs_src_-0a4558.js"></script>
<script src="/decidim-packs/js/decidim_core.js"></script>

```

so there is only one map js related call, not multiple.